cpp/src/community/k_truss_impl.cuh

```diff
@@ -293,63 +475,349 @@ k_truss(raft::handle_t const& handle,
+                          auto idx = thrust::distance(edgelist_first, itr_pair);
+                          edgelist_cnts[idx] -= decrease_count[i];
+                        }
+                       }
+                       );
+    
+        edges_to_mask.clear();
+        edges_to_mask.insert(edgelist_srcs.begin() + num_valid_edges,
+                             edgelist_srcs.end(),
+                             edgelist_dsts.begin() + num_valid_edges);
+        
+        // Remove weak edges in both direction from the undirected graph view
+        cugraph::transform_e(
+            handle,
+            cur_graph_view,
+            edges_to_mask,
+            cugraph::edge_src_dummy_property_t{}.view(),
+            cugraph::edge_dst_dummy_property_t{}.view(),
+            cugraph::edge_dummy_property_t{}.view(),
+            [] __device__(auto src, auto dst, thrust::nullopt_t, thrust::nullopt_t, thrust::nullopt_t) {
+              
+              return false;
+            },
+            weak_edges_mask.mutable_view(),
+            false);
+        
+        edges_to_mask.clear();
+        thrust::sort(
+          handle.get_thrust_policy(),
+          thrust::make_zip_iterator(edgelist_dsts.begin() + num_valid_edges, edgelist_srcs.begin() + num_valid_edges),
+          thrust::make_zip_iterator(edgelist_dsts.end(), edgelist_srcs.end())
+        );
+        
+        edges_to_mask.insert(edgelist_dsts.begin() + num_valid_edges,
+                             edgelist_dsts.end(),
+                             edgelist_srcs.begin() + num_valid_edges);
+        
+        cugraph::transform_e(
+            handle,
+            cur_graph_view,
+            edges_to_mask,
+            cugraph::edge_src_dummy_property_t{}.view(),
+            cugraph::edge_dst_dummy_property_t{}.view(),
+            cugraph::edge_dummy_property_t{}.view(),
+            [] __device__(auto src, auto dst, thrust::nullopt_t, thrust::nullopt_t, thrust::nullopt_t) {
+
+              return false;
+            },
+            weak_edges_mask.mutable_view(),
+            false);
+
+        edgelist_srcs.resize(num_valid_edges, handle.get_stream());
+        edgelist_dsts.resize(num_valid_edges, handle.get_stream());
+        edgelist_cnts.resize(num_valid_edges, handle.get_stream());
+
+        edgelist_first = thrust::make_zip_iterator(edgelist_srcs.begin(), edgelist_dsts.begin());
+        edgelist_last  = thrust::make_zip_iterator(edgelist_srcs.end(), edgelist_dsts.end());
+
+        if (prev_number_of_edges == cur_graph_view.compute_number_of_edges(handle)) { break; }
 
-      if (prev_number_of_edges == cur_graph_view.compute_number_of_edges(handle)) { break; }
     }
 
-    rmm::device_uvector<vertex_t> edgelist_srcs(0, handle.get_stream());
-    rmm::device_uvector<vertex_t> edgelist_dsts(0, handle.get_stream());
     std::optional<rmm::device_uvector<weight_t>> edgelist_wgts{std::nullopt};
-
+    
+    //#if 0
```

Comment:
Delete the commented out lines in this section

Reply:
Right I still had a lot of debug prints which are now removed